src/adapter.js

```diff
@@ -173,11 +177,36 @@ class Adapter {
       it = take(it, q.limit)
     }
 
-    if (q.keysOnly === true) {
-      return map(it, (e) => /** @type {Pair} */({ key: e.key }))
+    return it
+  }
+
+  /**
+   * @param {KeyQuery} q
```

Comment:
Any reason to not simply using the Query type?

Reply:
`Query` defines the Filter/Sort functions which use `Pair`/s as arguments and/or return types.  I thought about using generics to define the arg/return types of those methods but it seemed more straightforward to just have another type.